src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -296,78 +300,48 @@ private void validateOutputSettings() {
-        }
-
-        // ensure file exists
-        if (!new File(cvrPath).exists()) {
-          isValid = false;
-          Logger.log(Level.SEVERE, "Cast vote record file not found: %s", cvrPath);
-        }
-
-        // perform CDF checks
-        if (isCdf(source)) {
-          if (rawConfig.cvrFileSources.size() != 1) {
-            isValid = false;
-            Logger.log(Level.SEVERE, "CDF files must be tabulated individually.");
-          }
-          if (isTabulateByPrecinctEnabled()) {
-            isValid = false;
-            Logger.log(Level.SEVERE, "tabulateByPrecinct may not be used with CDF files.");
-          }
-        } else {
-          // perform ES&S checks
-
-          // ensure valid first vote column value
-          checkStringToIntWithBoundaries(source.getFirstVoteColumnIndex(), "firstVoteColumnIndex",
-              MIN_COLUMN_INDEX, MAX_COLUMN_INDEX, true, cvrPath);
-
-          // ensure valid first vote row value
-          checkStringToIntWithBoundaries(source.getFirstVoteRowIndex(), "firstVoteRowIndex",
-              MIN_ROW_INDEX, MAX_ROW_INDEX, true, cvrPath);
+      // ensure valid first vote column value
+      if (fieldOutOfRangeOrNotInt(source.getFirstVoteColumnIndex(), "firstVoteColumnIndex",
+          MIN_COLUMN_INDEX, MAX_COLUMN_INDEX, !isCdf(source), source.getFilePath())) {
+        sourceValid = false;
+      }
 
-          // ensure valid id column value
-          checkStringToIntWithBoundaries(source.getIdColumnIndex(), "idColumnIndex",
-              MIN_COLUMN_INDEX, MAX_COLUMN_INDEX, false, cvrPath);
+      // ensure valid first vote row value
+      if (fieldOutOfRangeOrNotInt(source.getFirstVoteRowIndex(), "firstVoteRowIndex",
+          MIN_ROW_INDEX, MAX_ROW_INDEX, !isCdf(source), source.getFilePath())) {
+        sourceValid = false;
+      }
 
-          // ensure valid precinct column value
-          checkStringToIntWithBoundaries(
-              source.getPrecinctColumnIndex(), "precinctColumnIndex", MIN_COLUMN_INDEX,
-              MAX_COLUMN_INDEX, false, cvrPath);
+      // ensure valid id column value
+      if (fieldOutOfRangeOrNotInt(source.getIdColumnIndex(), "idColumnIndex",
+          MIN_COLUMN_INDEX, MAX_COLUMN_INDEX, false, source.getFilePath())) {
+        sourceValid = false;
+      }
 
-          if (isNullOrBlank(source.getPrecinctColumnIndex()) && isTabulateByPrecinctEnabled()) {
-            isValid = false;
-            Logger.log(
-                Level.SEVERE,
-                "precinctColumnIndex is required when tabulateByPrecinct is enabled: %s",
-                cvrPath);
-          }
-        }
+      // ensure valid precinct column value
+      if (fieldOutOfRangeOrNotInt(
+          source.getPrecinctColumnIndex(), "precinctColumnIndex", MIN_COLUMN_INDEX,
+          MAX_COLUMN_INDEX, false, source.getFilePath())) {
+        sourceValid = false;
       }
     }
+    return sourceValid;
+  }
+
+  static boolean isCandidateValid(Candidate candidate) {
```

Comment:
`isCandidateNameValid`

This doesn't cover other ways in which the candidate might be invalid, e.g. a name or code collision or a missing code (when other candidates have codes).